pkg/cable/libreswan/libreswan.go

```diff
@@ -584,3 +607,55 @@ func (i *libreswan) runPluto() error {
 
 	return nil
 }
+
+func (i *libreswan) addClampMssToPathMTU(subnets string) {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := i.updateMangleTableEntryWithRetry(true, ruleSpecSource...); err != nil {
+		klog.Errorf("error appending iptables rule \"%s\": %v\n", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := i.updateMangleTableEntryWithRetry(true, ruleSpecDest...); err != nil {
+		klog.Errorf("error appending iptables rule \"%s\": %v\n", strings.Join(ruleSpecDest, " "), err)
+	}
+}
+
+func (i *libreswan) removeClampMssToPathMTU(subnets string) {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := i.updateMangleTableEntryWithRetry(false, ruleSpecSource...); err != nil {
+		klog.Errorf("error deleting iptables rule \"%s\": %v\n", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := i.updateMangleTableEntryWithRetry(false, ruleSpecDest...); err != nil {
+		klog.Errorf("error deleting iptables rule \"%s\": %v\n", strings.Join(ruleSpecDest, " "), err)
+	}
+}
+
+func (i *libreswan) updateMangleTableEntryWithRetry(add bool, rulespec ...string) error {
+	var err error
+
+	for retries := 1; retries < 5; retries++ {
+		if add {
+			if err = i.ipt.AppendUnique(mangleTable, subMarinerPostRoutingChain, rulespec...); err != nil {
+				time.Sleep(5 * time.Second)
+			}
+		} else {
+			if err = i.ipt.Delete(mangleTable, subMarinerPostRoutingChain, rulespec...); err != nil {
+				time.Sleep(5 * time.Second)
+			}
+		}
+
+		if err == nil {
+			break
+		}
+	}
```

Comment:
We could use `retry.OnError` here. However perhaps we should just propagate the error to the cable engine which will retry.

Reply:
Done.